iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/TieLineAdderImpl.java

```diff
@@ -70,76 +83,96 @@ public HalfLineAdderImpl setX(double x) {
         }
 
         @Override
-        public HalfLineAdderImpl setG1(double g1) {
-            this.g1 = g1;
+        public HalfLineAdderImpl setG(double g1) {
+            this.g = g1;
             return this;
         }
 
         @Override
-        public HalfLineAdderImpl setG2(double g2) {
-            this.g2 = g2;
+        public HalfLineAdderImpl setB(double b1) {
+            this.b = b1;
             return this;
         }
 
         @Override
-        public HalfLineAdderImpl setB1(double b1) {
-            this.b1 = b1;
+        public HalfLineAdderImpl setUcteXnodeCode(String ucteXnodeCode) {
+            this.halfUcteXnodeCode = ucteXnodeCode;
             return this;
         }
 
         @Override
-        public HalfLineAdderImpl setB2(double b2) {
-            this.b2 = b2;
-            return this;
+        public GenerationAdder<MergedDanglingLineAdder> newGeneration() {
+            return new GenerationAdderImpl<>(this);
         }
 
         @Override
         public TieLineAdderImpl add() {
-            if (id == null || id.isEmpty()) {
-                throw new ValidationException(this, String.format("id is not set for half line %d", num));
-            }
             if (Double.isNaN(r)) {
                 throw new ValidationException(this, String.format("r is not set for half line %d", num));
             }
             if (Double.isNaN(x)) {
                 throw new ValidationException(this, String.format("x is not set for half line %d", num));
             }
-            if (Double.isNaN(g1)) {
-                throw new ValidationException(this, String.format("g1 is not set for half line %d", num));
-            }
-            if (Double.isNaN(b1)) {
-                throw new ValidationException(this, String.format("b1 is not set for half line %d", num));
+            if (Double.isNaN(g)) {
+                throw new ValidationException(this, String.format("g is not set for half line %d", num));
             }
-            if (Double.isNaN(g2)) {
-                throw new ValidationException(this, String.format("g2 is not set for half line %d", num));
-            }
-            if (Double.isNaN(b2)) {
-                throw new ValidationException(this, String.format("b2 is not set for half line %d", num));
+            if (Double.isNaN(b)) {
+                throw new ValidationException(this, String.format("b is not set for half line %d", num));
             }
             if (num == 1) {
                 TieLineAdderImpl.this.halfLineAdder1 = this;
+                setNullableNode1(node);
+                setConnectableBus1(connectableBus);
+                setBus1(bus);
             }
```

Comment:
Another option is to use the following code, then the `setNullableNode1 `method it is not necessary and it can be removed.

  ```
  if (node != null) {
      setNode1(node);
  }
```

Reply:
No because we want it to be set to  `null` if the user makes a mistake in creating the half1 for example and want to "reset" it.

Reply:
ok